### PR TITLE
Feat: Sidebar logic

### PR DESCRIPTION
**Pull request by nicktruong**
### Tasks
- [Sidebar](https://trello.com/c/ffUhSXX9)
- [Home Page](https://trello.com/c/MxZy1vxh)

### Prerequisites
- https://github.com/nicktruong/chatbot-be/pull/3

### Describe
- Integrate sidebar with `create` chatbot api to create the bot's metadata
- Integrate sidebar with `getAll` chatbot api to get all bots and display on the sidebar
- Handle app navigation logic on component mount and on link click 

### Screenshots
image
image
image

### CODE REVIEW

File: src/hooks/apiHooks/mutations/useCreateBotMutation.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { UseMutationOptions, useMutation } from '@tanstack/react-query';
+
+import { axiosClient } from '@/apis';
+import { queryKeys } from '@/constants';
+
+import type { AxiosError } from 'axios';
```
Comment (carl-huynh-goldenowl) [inline src/hooks/apiHooks/mutations/useCreateBotMutation.ts:6]: Should place this import in line 2

File: src/interfaces/Bot.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export interface Bot {
+  id: string;
+  name: string;
+  createdAt: string;
+  updatedAt: string;
+  publishDate: string;
```
Comment (carl-huynh-goldenowl) [inline src/interfaces/Bot.ts:6]: Shoud we use publishDate in past simple?

File: src/app/pages/Home/components/SidebarContent/SidebarContent.tsx [hunk 3]
```diff
@@ -1,26 +1,76 @@
...
       </Box>
       <Box marginTop="1rem">
-        <NoMessagesIcon width="100%" />
-        <Text sx={styles.emptyText}>
-          {t(messages.thisWorkspaceDoesntHaveChatbotsYet)}
-        </Text>
+        {!!bots?.length ? (
+          bots.map(bot => {
+            const isActive = location.pathname.includes(bot.id);
+            return (
```
Comment (carl-huynh-goldenowl) [inline src/app/pages/Home/components/SidebarContent/SidebarContent.tsx:38]: Should write this as render function

File: src/app/pages/Home/components/SidebarContent/SidebarContent.tsx [hunk 4]
```diff
@@ -1,26 +1,76 @@
...
 
 export const SidebarContent = () => {
-  const { t } = usePrepareHook();
+  const { bots, location, isPending, t, onCreateBot } = usePrepareHook();
+
+  if (
+    bots?.length &&
+    [routes.home, routes.chatbot].includes(location.pathname)
+  ) {
+    return <Navigate to={routes.chatbotDetail.replace(':id', bots[0].id)} />;
```
Comment (carl-huynh-goldenowl) [inline src/app/pages/Home/components/SidebarContent/SidebarContent.tsx:21]: Should use `generatepath`
https://reactrouter.com/en/main/utils/generate-path

**Review by carl-huynh-goldenowl — CHANGES_REQUESTED** [review]

File: src/hooks/apiHooks/mutations/useCreateBotMutation.ts (hunk 1 above)
Comment (nicktruong) [reply src/hooks/apiHooks/mutations/useCreateBotMutation.ts:6]: Edited in: https://github.com/nicktruong/chatbot-fe/pull/11/files#diff-896bcbb7f72db0ab29472c743c383acf583b9ecc6c1c2b7b0ddfc7447bc343b9R1-R8

File: src/interfaces/Bot.ts (hunk 2 above)
Comment (nicktruong) [reply src/interfaces/Bot.ts:6]: I'll stick to `publishDate` as it is still correct and to be consistent with the backend side

File: src/app/pages/Home/components/SidebarContent/SidebarContent.tsx (hunk 3 above)
Comment (nicktruong) [reply src/app/pages/Home/components/SidebarContent/SidebarContent.tsx:38]: Edited in: https://github.com/nicktruong/chatbot-fe/pull/11/files#diff-40407a4ca35743b59214351056462342a5084b38006d2a1fc4077e232c3a8114R44-R85

File: src/app/pages/Home/components/SidebarContent/SidebarContent.tsx (hunk 4 above)
Comment (nicktruong) [reply src/app/pages/Home/components/SidebarContent/SidebarContent.tsx:21]: Edited in: https://github.com/nicktruong/chatbot-fe/pull/11/files#diff-2ba6268d786c9535754dbe0961b97df7447511ba69de5025c1ce62932efadc68R25-R27

**Review by carl-huynh-goldenowl — APPROVED** [review]

**Pull request merged**